src/gui/guiFormSpecMenu.cpp

```diff
@@ -1382,23 +1533,35 @@ void GUIFormSpecMenu::parseTabHeader(parserData* data, const std::string &elemen
 {
 	std::vector<std::string> parts = split(element,';');
 
-	if (((parts.size() == 4) || (parts.size() == 6)) ||
-		((parts.size() > 6) && (m_formspec_version > FORMSPEC_API_VERSION)))
+	if (((parts.size() == 4) || (parts.size() == 6)) || (parts.size() == 7 &&
+		data->real_coordinates) || ((parts.size() > 6) &&
```

Comment:
Adding a new parameter without increasing the formspec version will throw the error `Invalid TabHeader element ...` and become unusable on older clients. It sucks.

Reply:
With 7 parameters it's a new element. It is normal that new elements don't work with older clients. The old element is persisted, ergo this is ok imo.

Reply:
The dropdown does this as well in the geometry section by allowing height.  But if you notice in the code, both of them only allow the extra parameter if `real_coordinates` is enabled.

Breaking these elements doesn't even matter anyway because `real_coordinates` itself breaks forms anyway when put before the form size on older clients.  The clients think that size doesn't exist and spits out a funky FormSpec, as can be seen here:

![image](https://user-images.githubusercontent.com/31123645/57960401-5b742080-78bd-11e9-8cca-8e351d8f64f9.png)

I really wish we could just forget about backwards compatibility and just tell people to update.  But unfortunately, that's not how it works unless you're Microsoft.

Reply:
Right, it's guarded by the `real_coordinates` mode, more than that can't be done here.
It's fine to add new elements, as they don't make the entire formspec unusable. In this case it's a little bit different. The players might wonder why the formspec does not show up (correctly, if even). I have no idea on how to make this more user-friendly because node meta formspecs cannot be checked against the presence of `real_coordinates` to send the outdated client a message.
Leave it as-is. Players got to update.